frontend/app/utils/string-utils.ts

```diff
@@ -27,3 +27,13 @@ export function padWithZero(value: number, maxLength: number): string {
   if (value.toString().length >= maxLength) return value.toString();
   return value.toString().padStart(maxLength, '0');
 }
+
+/**
+ * Returns undefined for empty strings, whitespace, or undefined values.
+ * Otherwise, it returns the original string if it's not empty.
+ * @param str - The string to check, which may be undefined.
+ * @returns The original string or undefined.
+ */
+export function optionalString(str: string | undefined): string | undefined {
+  return !str || str.length === 0 || str.trim().length === 0 ? undefined : str;
```

Comment:
I think this can be written as: `return Number(str?.trim().length) > 0 ? str : undefined`, right?

Maybe consider adding a couple of tests to be sure.